### PR TITLE
Sample value creation

### PR DESCRIPTION
**Pull request by SimonHuber**
sample values defined on the one hand in builders and on the other hand in e.g. OffermanagementTestDataConstants.java

### CODE REVIEW

File: samples/core/src/test/java/io/oasp/gastronomy/restaurant/offermanagement/batch/impl/productimport/OfferManagementTest.java [hunk 1]
```diff
@@ -1,5 +1,9 @@
 package io.oasp.gastronomy.restaurant.offermanagement.batch.impl.productimport;
 
+import static io.oasp.gastronomy.restaurant.offermanagement.common.OffermanagementTestDataConstants.ID_OFFER_SCHNITZEL;
```
Comment (hohwille) [inline samples/core/src/test/java/io/oasp/gastronomy/restaurant/offermanagement/batch/impl/productimport/OfferManagementTest.java]: We should avoid static imports. Why not just qualify the constant where it is used? This makes the code also more readable. Managing and organizing static imports is pretty bad. Hence static imports are more an anti-pattern.

File: samples/core/src/test/java/io/oasp/gastronomy/restaurant/offermanagement/batch/impl/productimport/ProductManagementTest.java [hunk 2]
```diff
@@ -73,13 +77,13 @@ public void testFindProductsFiltered() {
...
     List<ProductEto> products = this.offerManagement.findProductsFiltered(filter, sort);
-    assertEquals(products.size(), 6);
+    assertEquals(products.size(), NUMBER_PRODUCT_MEAL);
 
     filter.setFetchDrinks(true);
     filter.setFetchMeals(false);
     sort.setSortByEntry(ProductSortByHitEntry.DESCRIPTION);
     products = this.offerManagement.findProductsFiltered(filter, sort);
-    assertEquals(products.size(), 4);
+    assertEquals(products.size(), NUMBER_PRODUCT_DRINK);
```
Comment (hohwille) [inline samples/core/src/test/java/io/oasp/gastronomy/restaurant/offermanagement/batch/impl/productimport/ProductManagementTest.java]: I really like this PR and it is a nice improvement over the ugly hardcoded magic values. But still the problem here is the test itself. Such assertions are ultra evil. First of all ```assertEquals``` is discouraged in OASP at all as we are using AssertJ what is much better and we need to get rid of any regular JUnit assert as well as hamcrest asserts. The sample is what should (some day) be considered as best practice and shows others how to do it right. Here we are doing things utterly wrong. The main issue is not the assertion syntax, though. It is asserting that the size of the result of a query has an expected magic value. This is very bad testing style. Instead you should assert the content of the result set and verify that particular hits are contained and that all containing hits fulfill the requested criteria.
BTW: I am totally aware that all this mess was not introduced by this PR. It is just that I am seeing this now watching the diff. We can file a new issue for that and it should not prevent accepting and merging this PR.

File: samples/core/src/test/java/io/oasp/gastronomy/restaurant/tablemanagement/service/impl/rest/TablemanagementRestServiceTest.java [hunk 3]
```diff
@@ -107,12 +106,11 @@ public void testDeleteTable() {
   public void testSaveTable() {
 
     // given
-    long tableNumber = 7L;
     long waiterId = 2L;
```
Comment (hohwille) [inline samples/core/src/test/java/io/oasp/gastronomy/restaurant/tablemanagement/service/impl/rest/TablemanagementRestServiceTest.java]: aint this a dead leftover that should also be removed?

File: samples/core/src/test/java/io/oasp/gastronomy/restaurant/tablemanagement/service/impl/rest/TablemanagementRestServiceTest.java [hunk 4]
```diff
@@ -134,10 +132,9 @@ public void testSaveTable() {
   public void testFindTablesByPost() {
 
     // given
-    long tableNumber = 7L;
     long waiterId = 2L;
```
Comment (hohwille) [inline samples/core/src/test/java/io/oasp/gastronomy/restaurant/tablemanagement/service/impl/rest/TablemanagementRestServiceTest.java]: see above and remove...

File: samples/core/src/test/java/io/oasp/gastronomy/restaurant/offermanagement/batch/impl/productimport/ProductManagementTest.java [hunk 5]
```diff
@@ -1,9 +1,9 @@
 package io.oasp.gastronomy.restaurant.offermanagement.batch.impl.productimport;
 
 import static io.oasp.gastronomy.restaurant.offermanagement.common.OffermanagementTestDataConstants.ID_PRODUCT_MEAL_SCHNITZEL;
-import static io.oasp.gastronomy.restaurant.offermanagement.common.OffermanagementTestDataConstants.NUMBER_PRODUCT_DRINK;
-import static io.oasp.gastronomy.restaurant.offermanagement.common.OffermanagementTestDataConstants.NUMBER_PRODUCT_MEAL;
-import static io.oasp.gastronomy.restaurant.offermanagement.common.OffermanagementTestDataConstants.NUMBER_PRODUCT_SIDEDISH;
+import static io.oasp.gastronomy.restaurant.offermanagement.common.OffermanagementTestDataConstants.NUMBER_OF_PRODUCT_DRINK;
+import static io.oasp.gastronomy.restaurant.offermanagement.common.OffermanagementTestDataConstants.NUMBER_OF_PRODUCT_MEAL;
+import static io.oasp.gastronomy.restaurant.offermanagement.common.OffermanagementTestDataConstants.NUMBER_OF_PRODUCT_SIDEDISH;
 import static org.junit.Assert.assertEquals;
```
Comment (hohwille) [inline samples/core/src/test/java/io/oasp/gastronomy/restaurant/offermanagement/batch/impl/productimport/ProductManagementTest.java]: I guess we need a rule for SonarQube and for Checkstyle/Eclipse that renderts a fat warning for ```org.junit.Assert```.

**Comment by kiran-vadla** [comment]
Changes submitted as part of this PR are submitted under a new branch under develop-2.3.0. PR  https://github.com/oasp/oasp4j/pull/529 is created for the same.

Thanks,
Kiran.

**Pull request closed without merging**